### ISSUE TITLE
Problem with HTTP Authentication

### ISSUE THREAD

**Issue by mmitch** [nilsbraden/ttrss-reader-fork#317]
I have just installed and configured tt-rss on my webserver – everything runs fine as far as I can see.
The apache configuration is SSL only (certificate by Let's Encrypt) and I've configured single-user mode without authentication in tt-rss, I want to do HTTP Authentication directly in apache.

I've installed TTRSS-Reader from the Google Play Store (version 1.89.0).

After enabling the api plugin in tt-rss, I can connect from TTRSS to my tt-rss instance as long as the HTTP Authentication on the server is disabled.

When I enable HTTP Authentication on the server, TTRSS can't connect to the API any more and gives the error message:
"401 (Message: Unauthorized)"

When I enable the HTTP Authentication setting in TTRSS and add a user and a password, I still get  the 401 error message.

According to my apache logs (via mod_log_debug), TTRSS never sends a username and password in the headers of the POST request, regardless of the "HTTP Authentication" setting in TTRSS.

Adding username/password directly to the tt-rss instance URL (like ``https://username:password@myserver.com/path/to/tt-rss/``) does not help either.

Did I miss some configuration option or is this a bug?

**Comment by ocdtrekkie** [comment]
I would go out on a limb to say "HTTP Authentication is currently broken in this app". Because I came here to report it, after a few of us have tested this app (both the F-Droid current, and GitHub current) against a working TTRSS with HTTP Auth, that works in the official Android app and a third party Windows app as well.

If @nilsbraden needs test credentials for an install that does not work with his app, I can provide them on request, I keep a dummy instance for this purpose.

The Sandstorm version of TTRSS may choose to temporarily delist this app as a recommended mobile client until this is confirmed to work. https://github.com/jparyani/Tiny-Tiny-RSS/pull/16

**Comment by mdewilde** [comment]
@ocdtrekkie I'd be willing to take a look at this if the issue is still open, and if you can provide me those test credentials.

**Comment by ocdtrekkie** [comment]
These should still be valid test credentials for a Sandstorm TTRSS instance with one feed on it, which requires HTTP Authentication to work.

Host: https://api-1afaa57ac327a243e9e49e045113fbe6.oasis.sandstorm.io
User: sandstorm
Password: RIt7H6ZT1CsZmu9Mt6zHcMucYK7wqEDsaKpRtJMYGbl

**Comment by mdewilde** [comment]
Thanks.
I'll let you know how it goes.

**Comment by mdewilde** [comment]
@ocdtrekkie I'm not having any luck with the instance. I get no reply on 
`curl -u sandstorm:RIt7H6ZT1CsZmu9Mt6zHcMucYK7wqEDsaKpRtJMYGbl -d '{"op":"login"}' http://api-1afaa57ac327a243e9e49e045113fbe6.oasis.sandstorm.io/api`
The site seems broken in a web browser.
Is this still a working instance?

**Comment by ocdtrekkie** [comment]
I am pretty certain @mdewilde. The URL is not accessible via a browser, this is normal. (If you're curious, you can try to create your own test at http://oasis.sandstorm.io on their free tier. Install TinyTinyRSS from https://apps.sandstorm.io/app/zj20q6pwy456cmq0k57n1mtqqtky664dfqnhsmf3t36khch5geph then start a TTRSS instance, go to Actions > Preferences > Mobile App and follow the instructions.)

I am pretty sure there is no Sandstorm-specific issues blocking the API endpoint though, because those credentials work with both the official Android app and a Windows Store app as well.

Also, did you try HTTPS? The credentials I provided were HTTPS, but your curl command appears to be HTTP.

**Comment by mdewilde** [comment]
Turns out the endpoint should be `api/` or `api/index.php` (and not `api`), which is why I wasn't getting a reply.
Now that I've managed that, fixing the issue shouldn't be all that hard. Without having checked the app yet, I believe it simply does not support single user mode yet (ie. it tries to send a user and password to TTRSS regardless of whether one is necessary/specified).
The relevant code is at https://github.com/nilsbraden/ttrss-reader-fork/blob/8d4aed5a85054edd053e2cb44d4809d1fc8e8c4a/ttrssreader/src/main/java/org/ttrssreader/net/JSONConnector.java#L469
I'll test and prepare a pull request later.
